### ISSUE TITLE
Getting a JobCancellationException on timeout or server not reachable

### ISSUE THREAD

**Issue by cnusp** [ktorio/ktor#1592]
version 1.3.0, client: apache

If a POST request is cancelled due to a timeout or server not reachable, a `JobCancellationException` is thrown. But it contains no information about the problem/cause.

1. Call an unreachable Service from within a `runBlocking` block.
2. JobCancellationException is thrown with message `Job was cancelled`, and stack trace having length 0.

**Expected behavior**
Meaningful exception which describes the actual network problem, and contains a non-empty stack trace.

**Comment by dtretyakov** [comment]
In 1.3.1 instead of actual exception `CancellationException` is thrown, but anyway it breaks existing code. The following test fails in 1.3.0/1.3.1:
```kt
    @Test
    fun `test ktor connect exception`() {
        assertFailsWith(ConnectException::class) {
            HttpClient(Apache).use { httpClient ->
                runBlocking {
                    httpClient.get<HttpResponse>("http://localhost:1234")
                }
            }
        }
    }
```

**Comment by OlSom** [comment]
Looks like it was fixed in 1.3.2. The test suggested by @dtretyakov  now passes using Ktor 1.3.2 and Kotlin 1.3.60

**Comment by e5l** [comment]
Sorry for the delay. Yep, the problem was fixed with the timeout feature release.
Closed
